Source/Core/Core/PowerPC/CachedInterpreter/CachedInterpreter.cpp

```diff
@@ -177,94 +89,89 @@ void CachedInterpreter::SingleStep()
   ExecuteOneBlock();
 }
 
-void CachedInterpreter::EndBlock(CachedInterpreter& cached_interpreter, UGeckoInstruction data)
+s32 CachedInterpreter::EndBlock(PowerPC::PowerPCState& ppc_state, const EndBlockOperands& operands)
 {
-  auto& ppc_state = cached_interpreter.m_ppc_state;
+  const auto& [downcount, num_load_stores, num_fp_inst] = operands;
   ppc_state.pc = ppc_state.npc;
-  ppc_state.downcount -= data.hex;
-  PowerPC::UpdatePerformanceMonitor(data.hex, 0, 0, ppc_state);
-}
-
-void CachedInterpreter::UpdateNumLoadStoreInstructions(CachedInterpreter& cached_interpreter,
-                                                       UGeckoInstruction data)
-{
-  PowerPC::UpdatePerformanceMonitor(0, data.hex, 0, cached_interpreter.m_ppc_state);
+  ppc_state.downcount -= downcount;
+  PowerPC::UpdatePerformanceMonitor(downcount, num_load_stores, num_fp_inst, ppc_state);
+  return 0;
 }
 
-void CachedInterpreter::UpdateNumFloatingPointInstructions(CachedInterpreter& cached_interpreter,
-                                                           UGeckoInstruction data)
+template <bool check_exceptions, bool write_pc>
+s32 CachedInterpreter::Interpret(PowerPC::PowerPCState& ppc_state,
+                                 const InterpretOperands<check_exceptions>& operands)
 {
-  PowerPC::UpdatePerformanceMonitor(0, 0, data.hex, cached_interpreter.m_ppc_state);
+  if constexpr (write_pc)
+  {
+    ppc_state.npc = (ppc_state.pc = operands.current_pc) + 4;
```

Comment:
Please write this in an easier-to-read way:

```suggestion
    ppc_state.pc = operands.current_pc;
    ppc_state.npc = operands.current_pc + 4;
```

Reply:
Unfortunately, writing it the way I have is more optimal. It skips getting the value of `operands.current_pc` a second time, saving one instruction.

Reply:
Did you observe this in some compiler, or multiple? Would putting `current_pc` in a local variable give you the same optimal codegen?

Reply:
Well now I can't reproduce it, but I could have sworn it made a difference in a different stage of development.